### PR TITLE
Lnx 162 add action to mine the stone

### PR DESCRIPTION
**Pull request by EasyJB**
Imo, its better to add interface, so any new action wchich will do same thing (destroy any object) will be easy to implement

### CODE REVIEW

File: lynx/common/actions/destroy_actions/chop.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from dataclasses import dataclass, field
+from typing import Dict
+
+from lynx.common.actions.destroy_actions.destroy_around import Destroy_around
+
+
+@dataclass
+class Chop(Destroy_around):
```
Comment (Krystian030) [inline lynx/common/actions/destroy_actions/chop.py]: change to `DestoryAround`

File: lynx/common/actions/destroy_actions/destroy_around.py [hunk 2]
```diff
@@ -1,40 +1,44 @@
...
+from lynx.common.actions.common_requirements import CommonRequirements
+from lynx.common.actions.create_object import CreateObject
+from lynx.common.actions.remove_object import RemoveObject
+from lynx.common.enums import Direction
+from lynx.common.object import Object
+from lynx.common.vector import Vector
+
+
+@dataclass
+class Destroy_around(Action):
```
Comment (Krystian030) [inline lynx/common/actions/destroy_actions/destroy_around.py]: Change class name

File: lynx/common/actions/destroy_actions/destroy_around.py [hunk 3]
```diff
@@ -1,40 +1,44 @@
...
+    """
+    Simple action used to hit/destroy object, which stands on the given destination.
+    """
+    object_id: int = -1
+    # target_position: Vector = Vector(1, 0) # Currently we want to chop the tree in specific direction
```
Comment (Krystian030) [inline lynx/common/actions/destroy_actions/destroy_around.py]: remove comments

**Review by Krystian030 — APPROVED** [review]
Gut

File: lynx/common/actions/destroy_actions/chop.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    _object_to_drop_after_destroyment: Dict[str, str] = field(default_factory=lambda: {"Tree": "Log"})
```
Comment (filip-szweda) [inline lynx/common/actions/destroy_actions/chop.py]: `_object_to_drop_after_destruction`

File: lynx/common/actions/destroy_actions/destroy_around.py [hunk 5]
```diff
@@ -1,40 +1,44 @@
...
+    direction: Vector = Direction.NORTH.value
+    # Dictionary, which has object name and what it will drop e.g. "
+    _object_to_drop_after_destroyment: Dict[int, str] = field(default_factory=dict)
```
Comment (filip-szweda) [inline lynx/common/actions/destroy_actions/destroy_around.py]: `_object_to_drop_after_destruction`

File: lynx/common/actions/destroy_actions/destroy_around.py [hunk 6]
```diff
@@ -1,40 +1,44 @@
...
+
+@dataclass
+class Destroy_around(Action):
+    """
+    Simple action used to hit/destroy object, which stands on the given destination.
+    """
+    object_id: int = -1
+    # target_position: Vector = Vector(1, 0) # Currently we want to chop the tree in specific direction
+    direction: Vector = Direction.NORTH.value
+    # Dictionary, which has object name and what it will drop e.g. "
```
Comment (filip-szweda) [inline lynx/common/actions/destroy_actions/destroy_around.py]: e.g. what?

File: lynx/common/actions/destroy_actions/destroy_around.py [hunk 7]
```diff
@@ -1,40 +1,44 @@
...
+    _object_to_drop_after_destroyment: Dict[int, str] = field(default_factory=dict)
+
+
+    def apply(self, scene: 'Scene') -> None:
+        target_position = self._get_target_position(scene)
+        objects_on_square = scene.get_objects_by_position(target_position)
+        x = self._object_to_drop_after_destroyment.keys
```
Comment (filip-szweda) [inline lynx/common/actions/destroy_actions/destroy_around.py]: What is this `x` for? Why is it named like that?

File: tests/common_requirements_test.py [hunk 8]
```diff
@@ -73,3 +73,16 @@ def test_multiple_success(self) -> NoReturn:
...
         assert not CommonRequirements.has_given_tags(self.scene, 123, ['walkable', 'pushable', 'insertable'])
+
+class TestIsAnyOnSquare:
+    scene = Scene()
+    dummy_object1 = Object(id=123, name="dummy", position=Vector(0, 0))
+    scene.add_entity(dummy_object1)
+    dummy_object2 = Object(id=1234, name="tree", position=Vector(0, 0))
+    scene.add_entity(dummy_object2)
+
+    def test_single_value_success(self) -> NoReturn:
```
Comment (filip-szweda) [inline tests/common_requirements_test.py]: `test_existing_tree_on_square_positive`

File: tests/common_requirements_test.py [hunk 9]
```diff
@@ -73,3 +73,16 @@ def test_multiple_success(self) -> NoReturn:
...
+        assert CommonRequirements.is_any_on_square(self.scene, Vector(0, 0), ['tree'])
+
+    def test_single_value_failure(self) -> NoReturn:
```
Comment (filip-szweda) [inline tests/common_requirements_test.py]: `test_nonexistent_object_on_square_negative`

**Review by filip-szweda — CHANGES_REQUESTED** [review]
Just some naming issues and odd variables. Otherwise LGTM

File: lynx/common/actions/destroy_actions/destroy_around.py (hunk 6 above)
Comment (EasyJB) [reply lynx/common/actions/destroy_actions/destroy_around.py]: if we want to Drop e.g. Logs for  tree and stone for rocks, or if we decide to drop anything else destruction of object

Comment (filip-szweda) [reply lynx/common/actions/destroy_actions/destroy_around.py]: Add to the comment please

**Review by filip-szweda — APPROVED** [review]

**Pull request merged**